docs/guides/db-connection-pooling.md

```diff
@@ -0,0 +1,151 @@
+---
+id: db-connection-pooling
+title: Database Connection Pooling
+sidebar_label: Database Connection Pooling
+---
+
+Opening a database connection is an expensive process, and that's why we use **[Connection pool](https://en.wikipedia.org/wiki/Connection_pool)** to reduce the cost. And it also allows us to have a good transaction management in a single session that uses different providers.
+
+The example uses PostgreSQL and **[node-postgres](https://node-postgres.com/features/transactions)** below;
+
+We define two providers in `DatabaseModule`, the first one is `Pool` which will be application scoped, and `DatabaseProvider` will be session-scoped. So, it will provide us different clients from connection pool for each session/network request. See Dependency Injection part of our docs to learn more about provider scopes.
```

Comment:
Please add here a note regarding the `Pool` class. Mention here that we can add external classes as `Provider`s and they will be declared in the application scope.